pkg/scalers/azure_data_explorer_scaler.go

```diff
@@ -0,0 +1,198 @@
+	client    *kusto.Client
+	name      string
+	namespace string
+}
+
+const adxName = "azure-data-explorer"
+
+var dataExplorerLogger = logf.Log.WithName("azure_data_explorer_scaler")
+
+func NewAzureDataExplorerScaler(config *ScalerConfig) (Scaler, error) {
+	metadata, err := parseAzureDataExplorerMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse azure data explorer metadata: %s", err)
+	}
+
+	client, err := azure.CreateAzureDataExplorerClient(metadata)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create azure data explorer client: %s", err)
+	}
+
+	return &azureDataExplorerScaler{
+		metadata:  metadata,
+		client:    client,
+		name:      config.Name,
+		namespace: config.Namespace,
+	}, nil
+}
+
+func parseAzureDataExplorerMetadata(config *ScalerConfig) (*azure.DataExplorerMetadata, error) {
+	metadata, err := parseAzureDataExplorerAuthParams(config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get database name.
+	databaseName, err := getParameterFromConfig(config, "databaseName", false)
+	if err != nil {
+		return nil, err
+	}
+	metadata.DatabaseName = databaseName
+
+	// Get endpoint.
+	endpoint, err := getParameterFromConfig(config, "endpoint", false)
+	if err != nil {
+		return nil, err
+	}
+	metadata.Endpoint = endpoint
+
+	// Get query.
+	query, err := getParameterFromConfig(config, "query", false)
+	if err != nil {
+		return nil, err
+	}
+	metadata.Query = query
+
+	// Get threshold.
+	if val, ok := config.TriggerMetadata["threshold"]; ok {
+		threshold, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing metadata. Details: can't parse threshold. Inner Error: %v", err)
+		}
+		metadata.Threshold = threshold
+	}
+
+	// Resolve metricName.
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		metadata.MetricName = kedautil.NormalizeString(fmt.Sprintf("%s-%s", adxName, val))
+	} else {
+		metadata.MetricName = GenerateMetricNameWithIndex(config.ScalerIndex, kedautil.NormalizeString(fmt.Sprintf("%s-%s", adxName, metadata.DatabaseName)))
+	}
+
+	dataExplorerLogger.Info("Parsed azureDataExplorerMetadata",
+		"database", metadata.DatabaseName,
+		"endpoint", metadata.Endpoint,
+		"metricName", metadata.MetricName,
+		"query", metadata.Query,
+		"threshold", metadata.Threshold)
+
+	return metadata, nil
+}
+
+func parseAzureDataExplorerAuthParams(config *ScalerConfig) (*azure.DataExplorerMetadata, error) {
+	metadata := azure.DataExplorerMetadata{}
+
+	switch config.PodIdentity {
+	case kedav1alpha1.PodIdentityProviderAzure:
+		metadata.PodIdentity = string(config.PodIdentity)
+	case "", kedav1alpha1.PodIdentityProviderNone:
+		// Check whether pod identity is provided through metadata instead of TriggerAuthentication object
+		podIdentity, err := getParameterFromConfig(config, "podIdentity", true)
+		if err != nil {
+			dataExplorerLogger.Info("pod identity is not provided. trying to resolve clientId, clientSecret and tenantId.")
```

Comment:
Do you think this is wort printing in INFO level? Maybe DEBUG would be better, like the other messages? WDYT?

Reply:
done